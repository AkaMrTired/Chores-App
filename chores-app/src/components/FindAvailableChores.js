import React from "react";
import PropTypes from "prop-types";
import ChoreCard from "./ChoreCard";

const FindAvailableChores = ({ chores }) => {
  const takeButton = () => {
    // this will assign the user ID to the chore.userId & update the chore.status to T
  };

  return (
    <div className="container">
      <h3>These chores are available!</h3>
      {/* mapping function to go through the chores list and render them where. */}
      {chores
        .filter((chore) => chore.status === "A")
        .map((chore) => (
          <ChoreCard
            key={chore._id}
            name={chore.name}
            price={chore.price}
            takeButton={takeButton}
          />
        ))}
      <div>
        <button type="button">
          <a href="/childdashboard">Back To Dashboard</a>
        </button>
      </div>
    </div>
  );
};

// we will need to edit the prop validation when we know what the data looks like.

FindAvailableChores.propTypes = {
  chores: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.number,
      name: PropTypes.string,
      price: PropTypes.string,
      status: PropTypes.string,
    })
  ).isRequired,
};
export default FindAvailableChores;
