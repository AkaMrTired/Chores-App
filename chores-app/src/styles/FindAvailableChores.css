.find-chore-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    background: var(--gradient);
    box-shadow: var(--shadow);
    padding: 3rem 1rem;
    border-radius: 20px;
    display: flex;
    flex-direction: column;
    width: 90vw;
    max-width: 1000px;
    margin: 5rem
  
}