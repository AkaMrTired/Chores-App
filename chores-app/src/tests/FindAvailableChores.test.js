import { render, screen } from "@testing-library/react";
import React from "react";
import FindAvailableChores from "../components/FindAvailableChores";

const validChores = [
  { _id: 123, name: "takenChore", price: "1", status: "T" },
  { _id: 1234, name: "availableChore", price: "1", status: "A" },
  { _id: 12345, name: "unavailableChore", price: "1", status: "U" },
];

describe("the parents dashboard", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<FindAvailableChores chores={validChores} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<FindAvailableChores chores={validChores} />);
    expect(screen.getByText(/these chores are available/i)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /take/i })).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /back to dashboard/i })
    ).toBeInTheDocument();
  });
  it("should render only the chores with the available status", () => {
    render(<FindAvailableChores chores={validChores} />);
    expect(screen.getByText(/availableChore/i)).toBeInTheDocument();
    expect(screen.queryByText(/unavailableChore/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/takenChore/i)).not.toBeInTheDocument();
  });
});
