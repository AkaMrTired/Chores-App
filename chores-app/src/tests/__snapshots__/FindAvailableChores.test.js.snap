// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the parents dashboard matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h3>
      These chores are available!
    </h3>
    <div>
      <div>
        availableChore
      </div>
      <div>
        Â£1
      </div>
      <button
        type="button"
      >
        Take
      </button>
    </div>
    <div>
      <button
        type="button"
      >
        <a
          href="/childdashboard"
        >
          Back To Dashboard
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
